### PR TITLE
eng: cleanup some leaks around around editors and workbenchInstantiationService

### PR DESCRIPTION
**Pull request by connor4312**
I added a disposable leak tracker to a test that used `workbenchInstantiationService`.
This fixes the baseline leaks and some extra leaks with that function.

### CODE REVIEW

**Review by DonJayamanne — DISMISSED** [review]

**Comment by bpasero** [comment]
@connor4312 this is very cool but should probably be reviewed by each respective component owner?

File: src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts [hunk 1]
```diff
@@ -146,6 +146,7 @@ export class InlineChatController implements IEditorContribution {
 	}
 
 	dispose(): void {
+		this._strategy?.dispose();
```
Comment (jrieken) [inline src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts:149]: I might be wrong but I don't believe this is needed, `finishExistingSession` should wind down everything which should also dispose the current startegy

File: src/vs/workbench/contrib/inlineChat/test/browser/inlineChatController.test.ts [hunk 2]
```diff
@@ -142,13 +141,12 @@ suite('InteractiveChatController', function () {
...
 
 	teardown(function () {
-		editor.dispose();
-		model.dispose();
 		store.clear();
 		ctrl?.dispose();
 	});
 
 	test('creation, not showing anything', function () {
+		for (let deadline = Date.now() + 1000; Date.now() < deadline;) { }
```
Comment (jrieken) [inline src/vs/workbench/contrib/inlineChat/test/browser/inlineChatController.test.ts:149]: what is that? busy loop to slow down testing?

**Review by jrieken — CHANGES_REQUESTED** [review]
request changes (or explanation) because of the busy loop in test

File: src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts (hunk 1 above)
Comment (connor4312) [reply src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts:149]: It appears this is needed, the leak detector pings if I remove the line. Maybe this is a bug in the test or elsewhere in the controller? Feel free to push changes!

**Comment by connor4312** [comment]
cc other owners

@joaomoreno @hediet @alexdima @sbatten 

please let me know if anything you own in this PR looks concerning

**Review by hediet — APPROVED** [review]

File: src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts:149]: Makes sense. The tests don't really await the (state machine's) run to terminate and I'd say it makes them to ugly to enforce that.

**Review by jrieken — APPROVED** [review]

**Comment by bpasero** [comment]
🙅 This needs to be reopened until I had a chance to review.

**Pull request merged**